lib/payout_pal/resource/token.rb

```diff
@@ -0,0 +1,63 @@
+module PayoutPal
+  module Resource
+    module Token
+
+      # If token is already expired, or will expire within
+      # `EXPIRATION_PADDING` time from now, refresh token.
+      EXPIRATION_PADDING = 2.freeze
+
+      def token
+        if expired?
+          time_request_was_issued = Time.now
+          @token = request_token
+          @expires_at = time_request_was_issued + @token.expires_in
+        end
+
+        @token
+      end
+
+
+      private
+
+      def expired?
+        Time.now >= (expires_at - EXPIRATION_PADDING)
```

Comment:
I feel like this might be tiny bit easier for me to follow if the 'padding' was added to `@expires_at` at the point it's being set, but I'm not sure why.